presto-main/src/main/java/io/prestosql/server/ui/FormWebUiAuthenticationManager.java

```diff
@@ -179,7 +192,48 @@ private static String encodeCurrentLocationForLoginRedirect(HttpServletRequest r
         return path;
     }
 
-    private void handleLoginRequest(HttpServletRequest request, HttpServletResponse response)
+    private void handleProtocolLoginRequest(HttpServletRequest request, HttpServletResponse response, FilterChain nextFilter)
+            throws IOException, ServletException
+    {
+        Authenticator authenticator = this.authenticator.orElseThrow(() -> new IllegalStateException("Authenticator is not present"));
```

Comment:
If you pass `authenticator` to this method, then you can call `authenticator.get()` above, inside the `if (authenticator.isPresent())` guard.